### PR TITLE
rsop: init at 0.3.9

### PR DESCRIPTION
**Pull request by nikstur**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Lassulus — APPROVED** [review]

**Review by Lassulus — CHANGES_REQUESTED** [review]
src fetcher gives a 502

File: pkgs/by-name/rs/rsop/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rsop";
+  version = "0.3.9";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "heiko";
+    repo = pname;
```
Comment (getchoo) [inline pkgs/by-name/rs/rsop/package.nix:17]:
```suggestion
    repo = "rsop";
```

File: pkgs/by-name/rs/rsop/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    hash = "sha256-K69vyZFaVvZj4yLaV/zQYoItvcTDuFR4mdmMcfl1UDA=";
+  };
+
+  cargoHash = "sha256-DJVgnfPpXf8hGX6Dv6W8GzqspMEFZHc2/Fkn1MZRXd0=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ pcsclite ];
+
+  passthru = {
```
Comment (getchoo) [inline pkgs/by-name/rs/rsop/package.nix:28]: A basic test might be nice
```suggestion
  passthru = {
    tests.version = testers.testVersion {
      command = "rsop version";
      package = rsop;
    };
```

File: pkgs/by-name/rs/rsop/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitea,
+  pkg-config,
+  pcsclite,
+  nix-update-script,
```
Comment (getchoo) [inline pkgs/by-name/rs/rsop/package.nix:7]:
```suggestion
  nix-update-script,
  testers,
  rsop
```

File: pkgs/by-name/rs/rsop/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    homepage = "https://codeberg.org/heiko/rsop";
+    description = "Stateless OpenPGP (SOP) based on rpgp";
+    license = with licenses; [
+      mit
+      apsl20
+      cc0
+    ];
+    maintainers = with lib.maintainers; [ nikstur ];
+    mainProgram = "rsop";
+  };
```
Comment (getchoo) [inline pkgs/by-name/rs/rsop/package.nix:42]: Nitpick, but would be nice as well
```suggestion
  meta = {
    homepage = "https://codeberg.org/heiko/rsop";
    description = "Stateless OpenPGP (SOP) based on rpgp";
    license = with lib.licenses; [
      mit
      apsl20
      cc0
    ];
    maintainers = with lib.maintainers; [ nikstur ];
    mainProgram = "rsop";
  };
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by getchoo** [comment]
> src fetcher gives a 502
```
$ nix-build -A rsop.src
/nix/store/aq6fhvg1qx14w63345j89mc5mcb8jx4p-source
```
Seems to work fine now. Maybe there was a codeberg outage?

**Comment by getchoo** [comment]
@ofborg eval

**Comment by nikstur** [comment]
@Lassulus merge

**Review by Lassulus — APPROVED** [review]

**Pull request merged**